### PR TITLE
[docs] Add a section for creating production builds locally

### PR DESCRIPTION
**Pull request by amandeepmittal**
# Why

Currently, in the Production build guide, we do not share info on how to generate production builds locally.

# How

This PR
- Adds a new section on how to create a production build locally and provide links to React Native docs for both Android and iOS.
- Also, splits the first section into two: What is a  production build and Production builds using EAS
- Fix verbiage issues as per our writing style guide
- Update BoxLink in Local app development guide to redirect to Production build locally section

**Preview**

CleanShot 2023-10-04 at 01 39 40

# Test Plan

Run docs locally and visit: http://localhost:3002/deploy/build-project/

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Review by Simek — APPROVED** [review]
Thanks for the updates, LGTM! 🚀

File: docs/pages/deploy/build-project.mdx [hunk 1]
```diff
@@ -12,11 +12,13 @@ import { BuildIcon, EasSubmitIcon, EasMetadataIcon, BookOpen02Icon } from '@expo
 
 Whether you have built a native app binary using [EAS](/build/setup/) or [locally](/guides/local-app-development/), the next step in your app development journey is to submit your app to the stores. To do so, you need to create a **production build**.
 
-## Production builds
+## What is a production build
```
Comment (brentvatne) [inline docs/pages/deploy/build-project.mdx:16]: i think we can delete the intro paragraph and "What is a production build" heading and jump right into the "Production builds are submitted to ..." paragraph

File: docs/pages/deploy/build-project.mdx [hunk 2]
```diff
@@ -12,11 +12,13 @@ import { BuildIcon, EasSubmitIcon, EasMetadataIcon, BookOpen02Icon } from '@expo
...
 
 Production builds are submitted to app stores for release to the general public or as part of a store-facilitated testing process such as TestFlight.
```
Comment (brentvatne) [inline docs/pages/deploy/build-project.mdx:17]:
```suggestion
Production builds are submitted to app stores for release to the general public or as part of a store-facilitated testing process such as TestFlight. This guide explains how to create production builds with [EAS](#anchor-goes-here) and [locally](#anchor-goes-here), but it's possible to create production builds for Expo apps with any CI service capable of compiling Android and iOS apps.
```

File: docs/pages/deploy/build-project.mdx [hunk 3]
```diff
@@ -92,6 +94,26 @@ By default, the `eas build` command will wait for your build to complete, but yo
 
 If you are a member of an organization and your build is on its behalf, you will find the build details on [the build dashboard for that account](https://expo.dev/accounts/[account]/builds).
 
+## Production builds locally
+
+To create a production build locally, see the following React Native guides for more information on the necessary steps that are required for Android and iOS.
```
Comment (brentvatne) [inline docs/pages/deploy/build-project.mdx:99]:
```suggestion
To create a production build locally, see the following React Native guides for more information on the necessary steps that are required for Android and iOS. Note that these guides assume your project has `android` and/or `ios` directories containing the respective native projects. If you use [Continuous Native Generation](/workflow/continuous-native-generation.mdx) then you will need to run [prebuild](/workflow/prebuild.mdx) to generate them before following these guides.
```

**Review by brentvatne — APPROVED** [review]
i'm unsure if we'll want to still with the "production builds" terminology (as opposed to "release builds") in the future, but let's not block on that today

**Comment by amandeepmittal** [comment]
@Brent, I've incorporated your suggestions/feedback.

> i'm unsure if we'll want to still with the "production builds" terminology (as opposed to "release builds") in the future, but let's not block on that today

Created a [linear issue]https://linear.app/expo/issue/ENG-10296/update-production-builds-terminology-to-release-builds) so to follow up later for now.

**Pull request merged**